docs/decisions/starter-kit-adr/KIT-ADR-0005-test-infrastructure-strategy.md

```
@@ -82,6 +82,69 @@ tests/
+
+**Problem**: Using `sys.exit(1)` at import time breaks pytest collection:
+
+```python
+# ❌ BAD: Breaks test collection when gql not installed
+try:
+    from gql import Client, gql
+except ImportError:
+    logger.error("gql package not installed")
+    sys.exit(1)  # Kills pytest during import!
+```
+
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)